app.rb

```diff
@@ -0,0 +1,158 @@
+require_relative 'teacher'
+
+class App
+  MENU_OPTIONS = {
+    '1' => :list_books,
+    '2' => :list_people,
+    '3' => :create_person,
+    '4' => :create_book,
+    '5' => :create_rental,
+    '6' => :list_all_rentals,
+    '7' => :exit_program
+  }.freeze
+
+  def initialize
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  def start
+    loop { choose_option }
+  end
+
+  def list_books
+    if @books.empty?
+      puts 'No books available. Please add some books and try again.'
+    else
+      puts "There are #{@books.count} book(s) available."
+      @books.each_with_index do |book, index|
+        puts "#{index + 1}) Title: \"#{book.title}\" | Author: #{book.author}"
+      end
+    end
+  end
+
+  def list_people
+    if @people.empty?
+      puts 'No people added yet. Please add a person and try again.'
+    else
+      @people.each_with_index do |person, index|
+        puts "#{index + 1}) [#{person.class}] Name: #{person.name} | Age: #{person.age} | ID: #{person.id}"
+      end
+    end
+  end
+
+  def create_person
+    print 'Would you like to create a student (1) or a teacher (2)? Select a number: '
+    choice = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    case choice
+    when '1'
+      create_student(name, age)
+    when '2'
+      create_teacher(name, age)
+    else
+      puts 'Invalid choice. Please try again.'
+    end
+  end
+
+  def create_student(name, age)
+    print 'Grade: '
+    grade = gets.chomp
+    print 'Parent permission? [Y/N]: '
+    permission_option = gets.chomp.downcase
+    student = Student.new(grade, name, age, parent_permission: permission_option == 'y')
+    @people << student
+    puts "Student created successfully. ID is #{student.id}"
+  end
+
+  def create_teacher(name, age)
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(specialization, name, age)
+    @people << teacher
+    puts "Teacher created successfully. Teacher ID is #{teacher.id}"
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+
+    if title.strip != '' && author.strip != ''
+      book = Book.new(title, author)
+      @books << book
+      puts 'Book created successfully.'
+    else
+      puts 'Please enter the book title and author.'
+    end
+  end
+
+  def create_rental
+    if @books.empty? || @people.empty?
+      puts 'Nothing to see here.'
+    else
+      puts 'Enter the number of the book you want: '
+      list_books
+      book_number = gets.chomp.to_i - 1
+      puts 'Enter your ID: '
+      list_people
+      person_id = gets.chomp.to_i
+      individual = @people.find { |person| person.id == person_id }
+
+      print 'Enter the date [yyyy-mm-dd]: '
+      date = gets.chomp.to_s
+      rental = Rental.new(date, @books[book_number], individual)
+      @rentals << rental
+      puts 'Book rented successfully.'
+    end
+  end
```

Comment:
- [ Required ] I think it would be nice if you fixed the error that occurs when a user tries to `create` a rental as suggested by the previous reviewer. I encountered the error below when trying to do so, please work on this and also ensure that the `list` rentals method also works as expected, I could not test it because the create rental method was not working.

## Error when creating a rental

![Screenshot from 2023-10-20 01-16-27](https://github.com/axcell-0/school-library/assets/16392046/10853691-f99f-49c9-bfcb-ecb2f4fc6cc7)